src/OSPSuite.Core/Domain/Services/PopulationSimulationPKAnalysesToDataTableConverter.cs

```diff
@@ -0,0 +1,85 @@
+      /// <param name="simulation">Simulation used to calculate the PK-Analyses</param>
+      /// <param name="pkAnalyses">Actual pkAnalyses to convert to <see cref="DataTable" /></param>
+      Task<DataTable> PKAnalysesToDataTableAsync(PopulationSimulationPKAnalyses pkAnalyses, IModelCoreSimulation simulation);
+
+      /// <summary>
+      ///    Creates a <see cref="DataTable" /> containing the PK-Analyses of the populationSimulation.
+      /// </summary>
+      /// <remarks>The format of the table will be IndividualId, QuantityPath, ParameterName, Value, Unit</remarks>
+      /// <param name="simulation">Simulation used to calculate the PK-Analyses</param>
+      /// <param name="pkAnalyses">Actual pkAnalyses to convert to <see cref="DataTable" /></param>
+      DataTable PKAnalysesToDataTable(PopulationSimulationPKAnalyses pkAnalyses, IModelCoreSimulation simulation);
+   }
+
+   public class PopulationSimulationPKAnalysesToDataTableConverter : IPopulationSimulationPKAnalysesToDataTableConverter
+   {
+      private readonly IDimensionFactory _dimensionFactory;
+      private readonly IPKParameterRepository _pkParameterRepository;
+
+      public PopulationSimulationPKAnalysesToDataTableConverter(
+         IDimensionFactory dimensionFactory, 
+         IPKParameterRepository pkParameterRepository)
+      {
+         _dimensionFactory = dimensionFactory;
+         _pkParameterRepository = pkParameterRepository;
+      }
+
+      public Task<DataTable> PKAnalysesToDataTableAsync(PopulationSimulationPKAnalyses pkAnalyses, IModelCoreSimulation simulation)
+      {
+         return Task.Run(() => PKAnalysesToDataTable(pkAnalyses, simulation));
+      }
+
+      public DataTable PKAnalysesToDataTable(PopulationSimulationPKAnalyses pkAnalyses, IModelCoreSimulation simulation)
+      {
+         var dataTable = new DataTable(simulation.Name);
+
+         dataTable.AddColumn<int>(INDIVIDUAL_ID);
+         dataTable.AddColumn<string>(QUANTITY_PATH);
+         dataTable.AddColumn<string>(PARAMETER);
+         dataTable.AddColumn<string>(VALUE);
+         dataTable.AddColumn<string>(UNIT);
+
+         dataTable.BeginLoadData();
+         foreach (var quantityPKParameter in pkAnalyses.All())
+         {
+            var quantityPath = quantityPKParameter.QuantityPath;
+            var molWeight = simulation.MolWeightFor(quantityPath);
+            var pkParameter = _pkParameterRepository.FindByName(quantityPKParameter.Name);
+            var quantityPKParameterContext = new QuantityPKParameterContext(quantityPKParameter, molWeight);
+            var mergedDimension = _dimensionFactory.MergedDimensionFor(quantityPKParameterContext);
+            var unit = mergedDimension.UnitOrDefault(pkParameter.DisplayUnitName);
+            quantityPKParameter.Values.Each((value, index) =>
+            {
+               var row = dataTable.NewRow();
+               row[INDIVIDUAL_ID] = index;
```

Comment:
@Yuri05 This might be wrong here. We are using the index instead of individualId. This was always like this so this is not new but we should think about what it means